src/main/flight/pid.c

```diff
@@ -253,21 +253,77 @@ void pidInitConfig(const pidProfile_t *pidProfile) {
     ITermWindupPointInv = 1.0f / (1.0f - ITermWindupPoint);
 }
 
-static float calcHorizonLevelStrength(void) {
-    float horizonLevelStrength = 0.0f;
-    if (horizonTransition > 0.0f) {
-        const float mostDeflectedPos = MAX(getRcDeflectionAbs(FD_ROLL), getRcDeflectionAbs(FD_PITCH));
-        // Progressively turn off the horizon self level strength as the stick is banged over
-        horizonLevelStrength = constrainf(1 - mostDeflectedPos * horizonTransition, 0, 1);
+// calculates strength of horizon leveling; 0 = none, 1.0 = most leveling
+static float calcHorizonLevelStrength(const pidProfile_t *pidProfile) {
+    // start with 1.0 at center stick, 0.0 at max stick deflection:
+    float horizonLevelStrength = 1.0f -
+             MAX(getRcDeflectionAbs(FD_ROLL), getRcDeflectionAbs(FD_PITCH));
+
+    // 0 at level, 90 at vertical, 180 at inverted (degrees):
+    const float currentInclination = MAX(ABS(attitude.values.roll),
+                                        ABS(attitude.values.pitch)) / 10.0f;
+
+    // horizonTiltMode:  SAFE = leveling always active when sticks centered,
+    //                   EXPERT = leveling can be totally off when inverted
+    if (pidProfile->horizon_tilt_mode == HORIZON_TILT_MODE_EXPERT) {
+        if (pidProfile->horizon_tilt_effect < 175) {
+            // horizonTiltEffect 0 to 125 => 270 to 90
+            //  (represents where leveling goes to zero):
+            const float cutoffDegrees = (175-pidProfile->horizon_tilt_effect) * 1.8f;
+            // inclinationLevelRatio (0.0 to 1.0) is smaller (less leveling)
+            //  for larger inclinations; 0.0 at cutoffDegrees value:
+            const float inclinationLevelRatio = constrainf(
+                       (cutoffDegrees-currentInclination) / cutoffDegrees, 0, 1);
+            // apply configured horizon sensitivity:
+            if (horizonTransition <= 0) {        // zero means no leveling
+                horizonLevelStrength = 0;
+            } else {
+                // when stick is near center (horizonLevelStrength ~= 1.0)
+                //  H_sensitivity value has little effect,
+                // when stick is deflected (horizonLevelStrength near 0.0)
+                //  H_sensitivity value has more effect:
+                horizonLevelStrength = (horizonLevelStrength - 1) *
+                        100 / horizonTransition + 1;
+            }
+            // apply inclination ratio, which may lower leveling
+            //  to zero regardless of stick position:
+            horizonLevelStrength *= inclinationLevelRatio;
+        }
+        else
+          horizonLevelStrength = 0;
+    } else {  // horizon_tilt_mode = SAFE (leveling always active when sticks centered)
+        float sensitFact;
+        if (pidProfile->horizon_tilt_effect > 0) {
+            // 0 to 100 => 1.0 to 0.0 (larger means more leveling):
+            const float factorRatio = (float)(100 - pidProfile->horizon_tilt_effect) / 100;
```

Comment:
There are the two cases (EXPERT/SAFE), so pre-loading 'factorRatio' would only apply to one.  How about if we change the "/100" to a "*0.01f"?  (If you really want pre-loading then we'd want to do it for both 'cutoffDegrees' and 'factorRatio' -- let me know if that's your perference and I'll implement it that way.  We could also pre-load two different versions of 'horizonTransition'.  It's a question of: how much pre-loading of how many variables do we want to do?)

I've done a commit to change the "/100" to a "*0.01f" and to do a minor improvement in the horizon_tilt_mode=EXPERT case.